### PR TITLE
feat: Timer Solution

### PR DESCRIPTION
**Pull request by madvosburg**
Solution to debouncing using timer

### CODE REVIEW

File: Core/Src/main.c [hunk 1]
```diff
@@ -116,35 +122,35 @@ int main(void)
...
+		    sendData(TxData);
+		    TxData[0] = 0;
+		    flag = 0;
+		  }
+		 if (flag == 3)
+		 {
+		    TxData[0] = 3;
+		    sendData(TxData);
+		    TxData[0] = 0;
+		    flag = 0;
```
Comment (alextran1502) [inline Core/Src/main.c:145]: What do these flag values mean? Maybe create an `enum` for each of the flag value?

File: Core/Src/main.c [hunk 2]
```diff
@@ -212,9 +249,9 @@ static void MX_TIM16_Init(void)
 
   /* USER CODE END TIM16_Init 1 */
   htim16.Instance = TIM16;
-  htim16.Init.Prescaler = 79;
+  htim16.Init.Prescaler = 19;
   htim16.Init.CounterMode = TIM_COUNTERMODE_UP;
-  htim16.Init.Period = 65535;
+  htim16.Init.Period = 41999;
```
Comment (alextran1502) [inline Core/Src/main.c:254]: How many `ms` does the timer get triggered?

Comment (madvosburg) [reply Core/Src/main.c:254]: Should be 10 ms

**Review by alextran1502 — APPROVED** [review]
The code structure is clear and well done. Need to verify some values to increase the readability of the code

File: Core/Src/main.c (hunk 2 above)
Comment (alextran1502) [reply Core/Src/main.c:254]: Can you include how it is calculated as a comment?

File: Core/Src/main.c (hunk 1 above)
Comment (madvosburg) [reply Core/Src/main.c:145]: Yeah I can definitely make those more organized and clear. They keep track of what button was pressed but I understand that isn't obvious

**Pull request merged**